dockerfiles/Dockerfile.android-builder

```diff
@@ -0,0 +1,53 @@
+FROM openjdk:8-jdk-stretch
+
+RUN apt-get update
+RUN apt-get install -y gradle jq
+
+WORKDIR /sdk
+RUN wget https://dl.google.com/android/repository/sdk-tools-linux-4333796.zip
+RUN unzip sdk-tools-linux-4333796.zip
+
+ENV PATH="${PATH}:/sdk/tools:/sdk/tools/bin"
+
+RUN yes | sdkmanager "platform-tools"
+
+ENV PATH="${PATH}:/sdk/platform-tools"
+ENV ANDROID_HOME="/sdk/"
+
+WORKDIR /app
+
+COPY gradlew build.gradle settings.gradle gradle.properties /app/
+COPY gradle/ /app/gradle/
+COPY sdk/ sdk/
+COPY scripts/ scripts/
+
+RUN scripts/install-ndk.sh
+
+RUN ./gradlew
+
+# Everything above this point should be derived from android-base
```

Comment:
With correct caching setup it shouldn't actually matter if it runs directly from the android-base image, as we can just cache the duplicate layers from there.  I prefer that personally as it'll be easier to run locally.

Reply:
Maintainability is more the issue - if the two Dockerfiles have subtle differences and need to duplicate steps then that's fine, but otherwise it'd be good only having to make a change in one place.